.github/workflows/check_security_vulnerability.yml

```diff
@@ -1,32 +1,82 @@
-# This workflow uses actions that are not certified by GitHub.
-# They are provided by a third-party and are governed by
-# separate terms of service, privacy policy, and support
-# documentation.
-
-name: Check Security Vulnerability
+name: Code Quality and Security
 
 on:
   pull_request:
   push:
     branches:
       - main
+  schedule:
+    - cron: '0 0 * * 0' # Run every Sunday at 00:00 (midnight)
 
 jobs:
-  lint:
-    name: DevSkim
-    runs-on: ubuntu-latest
+  shared-setup:
+    name: Shared Setup
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
     permissions:
       actions: read
       contents: read
       security-events: write
+    outputs:
+      checkout_ref: ${{ steps.checkout.outputs.ref }}
+      os: ${{ matrix.os }}
     steps:
       - name: Checkout code
+        id: checkout
         uses: actions/checkout@v4
 
+  devskim:
+    name: DevSkim Security Scan
+    needs: shared-setup
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.shared-setup.outputs.checkout_ref }}
       - name: Run DevSkim scanner
         uses: microsoft/DevSkim-Action@v1
-
       - name: Upload DevSkim scan results to GitHub Security tab
         uses: github/codeql-action/upload-sarif@v3
         with:
           sarif_file: devskim-results.sarif
+
+  rust-clippy:
+    name: Rust Clippy Analysis
+    needs: shared-setup
+    runs-on: ${{ needs.shared-setup.outputs.os }}
```

Comment:
From the CI output, this job was only executed on [Windows](https://github.com/topgrade-rs/topgrade/actions/runs/10127215929/job/28004716053), any idea?

> info: syncing channel updates for 'stable-x86_64-pc-windows-msvc'

Reply:
Are you referring to https://github.com/topgrade-rs/topgrade/actions/runs/10402662730 ?

Reply:
Or https://github.com/topgrade-rs/topgrade/actions/runs/10402662748 ?

Reply:
> Or https://github.com/topgrade-rs/topgrade/actions/runs/10402662748 ?

This and the CI referred in my comment both run the clippy only on Windows

Reply:
I guess

> ${{ needs.shared-setup.outputs.os }}

this is not working?

Reply:
> > Or https://github.com/topgrade-rs/topgrade/actions/runs/10402662748 ?
> 
> This and the CI referred in my comment both run the clippy only on Windows

I do not see it. Do you have the direct link to the output or provide a screenshot?

Reply:
![image](https://github.com/user-attachments/assets/ef199667-1477-4bef-bf97-d3b01434131d)

See the "install toolchain" step, it is running on Windows